### PR TITLE
Fix `p7zip_jll` environment setup

### PR DESCRIPTION
**Pull request by staticfloat**
On most platforms, setting `LD_LIBRARY_PATH` and friends was not
necessary for invoking `7z` because we fell back to the natural library
search paths for things like `libstdc++`, but in truly minimal
environments (like Alpine linux) this does not work; so let's set these
paths properly.  We do this work manually in v1.6, in v1.7 we hope to
have a JLLWrappers-based solution.

### CODE REVIEW

**Comment by stevengj** [comment]
IIRC there is a way to set the environment for spawning a Cmd without modifying the global ENV.

**Comment by staticfloat** [comment]
There is, I added it. :)

this provides both ways of doing it, for backwards compatibility.

File: stdlib/p7zip_jll/src/p7zip_jll.jl [hunk 1]
```diff
@@ -21,9 +21,51 @@ else
...
+                env["PATH"] = string(PATH, pathsep, env["PATH"])
+            else
+                env["PATH"] = PATH
+            end
+        end
+    end
+    return env
+end
+
+function p7zip(f::Function; adjust_PATH::Bool = true, adjust_LIBPATH::Bool = true)
```
Comment (stevengj) [inline stdlib/p7zip_jll/src/p7zip_jll.jl:59]: Should this method just be deprecated?

File: stdlib/p7zip_jll/src/p7zip_jll.jl [hunk 2]
```diff
@@ -21,9 +21,51 @@ else
...
+    const pathsep = ';'
+elseif Sys.isapple()
+    const LIBPATH_env = "DYLD_FALLBACK_LIBRARY_PATH"
+    const LIBPATH_default = "~/lib:/usr/local/lib:/lib:/usr/lib"
+    const pathsep = ':'
+else
+    const LIBPATH_env = "LD_LIBRARY_PATH"
+    const LIBPATH_default = ""
+    const pathsep = ':'
+end
```
Comment (stevengj) [inline stdlib/p7zip_jll/src/p7zip_jll.jl:36]: Should this definition be in Base somewhere?

File: stdlib/p7zip_jll/src/p7zip_jll.jl (hunk 1 above)
Comment (Keno) [reply stdlib/p7zip_jll/src/p7zip_jll.jl:59]: Or maybe we can pass an appropriate `Cmd` object, rather than the string?

Comment (Keno) [reply stdlib/p7zip_jll/src/p7zip_jll.jl:59]: I guess that doesn't buy us anything over the other method.

Comment (staticfloat) [reply stdlib/p7zip_jll/src/p7zip_jll.jl:59]: We could deprecate it, but we're in a bit of an ecosystem issue in that if a package, which uses JLLs, wants to support 1.3-1.6, they will then get deprecation warnings on 1.6 for using the older, compatible interface.  I think I would rather let things be for a bit, and then start introducing deprecation warnings once we have the data showing that we have a majority of users on 1.6+, and that package devs can actually switch over without annoying too many users.

File: stdlib/p7zip_jll/src/p7zip_jll.jl (hunk 2 above)
Comment (staticfloat) [reply stdlib/p7zip_jll/src/p7zip_jll.jl:36]: Yeah, probably.  In v1.7 this is all going to be generated by a stdlib-vendored copy of JLLWrappers, this is all just copied from there for the interim.

**Comment by giordano** [comment]
I guess this should be backported for v1.6?

**Pull request merged**